posts/2021-03-13-microprofile-config-2.0.adoc

```diff
@@ -104,36 +105,47 @@ The examples below use the config property values defined in this example META-I
 
-Where the `-Dliberty.var.mp.config.profile` argument sets a link:https://maven.apache.org/pom.html#Properties[Maven property] which the liberty-maven-plugin link:https://github.com/scottkurz/ci.maven/blob/f3920800351b6d2c26e62a19008b68093afa48ea/docs/common-server-parameters.md#setting-liberty-configuration-with-maven-project-properties[uses to configure the Liberty app]. In this case, `mp.config.profile` property is set to `testing`.
+If you deploy your microservices to Kubernetes, you can set the property either:
 
-`mp.config.profile` can be used:
+- In a ConfigMap, which then maps to an environment variable (Refer to link:https://openliberty.io/guides/kubernetes-microprofile-config.html#creating-a-configmap-and-secret[this blog] for more information on how to use ConfigMap to store properties); or
+- Directly as an environment variable in your deployment YAML file.  
+
+When specifying this property as an environment variable, you should use `MP_CONFIG_PROFILE` so that it works on all Operating Systems.
+
+The `mp.config.profile` property can be used:
 
 - link:https://download.eclipse.org/microprofile/microprofile-config-2.0/microprofile-config-spec-2.0.html#_on_property_level[On property level]: config property names can be set in the following format so that they are used for specific selected profiles: 
 +
  %<mp.config.profile>.<original property name> 
 +
 For example, with `mp.config.profile` set to `testing`, retrieving the config value for "server.host" would use the config property `%testing.server.host` from the <<example-ConfigSource, example ConfigSource>> rather than `server.host`. The value of the property would resolve to `example.test.org`.
 +
-If `mp.config.profile` was not set, retrieving "server.host" would resolve to `example.org`.
+Similarly, if `mp.config.profile` was set to `production`, retrieving "server.host" would resolve to `example.prod.org`. If `mp.config.profile` was **not** set, retrieving "server.host" would resolve to `example.org`.
 
 - link:https://download.eclipse.org/microprofile/microprofile-config-2.0/microprofile-config-spec-2.0.html#_on_config_source_level[On ConfigSource level]: multiple microprofile-config.properties files can be provided in the following format so they can be used for specific selected profiles:
 +
  microprofile-config-<mp.config.profile>.properties 
 +
 For example, if a file called microprofile-config-testing.properties was provided on the classpath, with `mp.config.profile` set to `testing` the file would be loaded "on top of" the default microprofile-config.properties file. The config property values from microprofile-config-testing.properties would take precedence.
```

Comment:
Change,
"The config property values from microprofile-config-testing.properties would take precedence."
to,
"The config property values from microprofile-config-testing.properties take precedence."

Reply:
I'm not sure about this one. The behaviour is dependent on the condition from the previous sentence.